robot_calibration/src/nodes/calibrate_server.cpp

```diff
@@ -0,0 +1,28 @@
+#include <ctime>
+#include <rclcpp/rclcpp.hpp>
+#include "rclcpp_action/rclcpp_action.hpp"
+
+#include <robot_calibration/util/action_server.hpp>
+
+int main(int argc, char **argv)
+{
+  rclcpp::init(argc, argv);
+  rclcpp::Node::SharedPtr node = std::make_shared<rclcpp::Node>("robot_calibration_server");
+
+  // Get calibration poses YAML path
+  std::string data_source("calibration_poses.yaml");
+  if (argc > 1)
+  {
+    data_source = argv[1];
+  }
+
+  std::cout << "Creating action server" << std::endl;
+  CalibratePoseServer calibrate_pose_action_server = CalibratePoseServer(node, data_source);
+
```

Comment:
Could you add this comment here?
`// TODO: This's a workaround a bug in MultiThreadedExecutor deadlock when spinning without a timeout`

Reply:
What is the difference between `rclcpp::spin(node)` and using a MultiThreadedExecutor? Both seems to work fine

Reply:
In the code below we add the node (CalibratePoseServer) to the multi-threaded executor which is capable of spinning a multi-threaded node (when built correctly the CalibratePoseServer can create as many thread as it requires to listen for topics/services/actions).

When you call `spin(node)` you are calling in on the node so it depends what type of node you have built. If it node has listeners configured with:
`node->create_callback_group(rclcpp::CallbackGroupType::MutuallyExclusive)` or
`node->create_callback_group(rclcpp::CallbackGroupType::Reentrant)`
then the nodes executor needs to support multi-threading.